src/substrate/useSubstrate.js

```diff
@@ -0,0 +1,64 @@
+import { useContext, useEffect, useCallback } from 'react';
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { web3Accounts, web3Enable } from "@polkadot/extension-dapp";
+import keyring from "@polkadot/ui-keyring";
+
+import config from "../config";
+import { SubstrateContext } from './SubstrateContext';
+
+const useSubstrate = () => {
+  let [state, dispatch] = useContext(SubstrateContext);
+
+  // `useCallback` so that returning memoized function and not created
+  //   everytime, and thus re-render.
+  const connect = useCallback(async() => {
+    let { api, socket, types } = state;
+    if (api) return;
+
+    const provider = new WsProvider(socket);
+
+    try {
+      api = await ApiPromise.create({ provider, types });
+      dispatch({ type: 'CONNECT', payload: api });
+      api.isReady.then(() => dispatch({type: 'CONNECT_SUCCESS'}));
+
+    } catch(e) {
+      console.log(e);
+      dispatch({ type: 'CONNECT_ERROR' });
+    }
+    // eslint-disable-next-line
```

Comment:
This is suppressinga warning
```
Line 29:  React Hook useCallback has missing dependencies: 'dispatch' and 'state'. Either include them or remove the dependency array               react-hooks/exhaustive-deps
```
Is disabling the correct way to remove that warning? If so, help me understand why.

Reply:
need to be further discussed and I am doing it in discretion. For `connect`, dispatch is just a dispatcher function of `useReducer` that is never changed, and `state` has so many properties inside. It really doesn't make sense to put the whole state in.

Reply:
fixed :)